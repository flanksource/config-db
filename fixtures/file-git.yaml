apiVersion: configs.flanksource.com/v1
kind: ScrapeConfig
metadata:
  name: file-git-scraper
spec:
  file:
    - type: $.kind
      id: $.metadata.name
      url: github.com/flanksource/canary-checker
      paths:
        - fixtures/minimal/http_pass_single.yaml
