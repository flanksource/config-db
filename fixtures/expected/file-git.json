[
  {
    "last_modified": "0001-01-01T00:00:00Z",
    "source": "github.com/flanksource/canary-checker/fixtures/minimal/http_pass_single.yaml",
    "id": "http-pass-single",
    "config_class": "Canary",
    "config": {
      "apiVersion": "canaries.flanksource.com/v1",
      "kind": "Canary",
      "metadata": {
        "labels": {
          "canary": "http"
        },
        "name": "http-pass-single"
      },
      "spec": {
        "interval": 30,
        "http": [
          {
            "endpoint": "https://httpbin.demo.aws.flanksource.com/status/200",
            "name": "http-deprecated-endpoint"
          },
          {
            "name": "http-minimal-check",
            "url": "https://httpbin.demo.aws.flanksource.com/status/200",
            "metrics": [
              {
                "name": "httpbin_2xx_count",
                "type": "counter",
                "value": "code == 200 ? 1 : 0",
                "labels": [
                  {
                    "name": "name",
                    "value": "httpbin_2xx_count"
                  },
                  {
                    "name": "check_name",
                    "valueExpr": "check.name"
                  },
                  {
                    "name": "status_class",
                    "valueExpr": "string(code).charAt(0)"
                  }
                ]
              }
            ]
          },
          {
            "name": "http-param-tests",
            "url": "https://httpbin.demo.aws.flanksource.com/status/200",
            "responseCodes": [201, 200, 301],
            "responseContent": "",
            "maxSSLExpiry": 7
          },
          {
            "name": "http-expr-tests",
            "url": "https://httpbin.demo.aws.flanksource.com/status/200",
            "test": {
              "expr": "code in [200,201,301] && sslAge > Duration('7d')"
            },
            "display": {
              "template": "code={{.code}}, age={{.sslAge}}"
            }
          },
          {
            "name": "http-headers",
            "test": {
              "expr": "json.headers[\"User-Agent\"].startsWith(\"canary-checker/\")"
            },
            "url": "https://httpbin.demo.aws.flanksource.com/headers"
          }
        ]
      }
    }
  }
]
